src/Vector3.php

```diff
@@ -33,56 +33,57 @@
 use function sqrt;
 use const PHP_ROUND_HALF_UP;
 
-class Vector3{
+readonly class Vector3{
+
 	public function __construct(
-		public float|int $x,
-		public float|int $y,
-		public float|int $z
+		protected float|int $x,
+		protected float|int $y,
+		protected float|int $z
```

Comment:
I don't like this change. The whole point of making the class readonly is to allow direct access to its fields without mutability issues.